tests/e2e/performance/test_small_file_workload.py

```diff
@@ -5,24 +5,310 @@
 import pytest
 import time
 
-from ocs_ci.ocs.ocp import OCP
-from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs.ocp import (OCP, get_ocp_clustername, get_ocs_version,
+                            get_ocp_build, get_ocp_channel, get_ocp_provider)
+from ocs_ci.utility.utils import TimeoutSampler, get_ocp_version, run_cmd
 from ocs_ci.ocs.resources.ocs import OCS
 from ocs_ci.utility import templating
 from ocs_ci.ocs.utils import get_pod_name_by_pattern
 from ocs_ci.ocs.ripsaw import RipSaw
 from ocs_ci.ocs import constants
 from ocs_ci.framework.testlib import E2ETest, performance
 from tests.helpers import get_logs_with_errors
+from elasticsearch import (Elasticsearch, exceptions as ESExp)
+import numpy as np
 
 log = logging.getLogger(__name__)
 
 
+class SmallFileResultsAnalyse(object):
+    """
+    This class is reading all test results from elasticsearch server (which the
+    ripsaw running of the benchmark is generate), aggregate them by :
+        test operation (e.g. create / delete etc.)
+          sample (for test to be valid it need to run with more the one sample)
+            host (test can be run on more then one pod {called host})
+
+    it generate results for the all test as one unit which will be valid only
+    if the deviation between samples is less the 5%
+
+    """
+
+    managed_keys = {
+        'IOPS': {'name': 'iops', 'op': np.sum},
+        'MiBps': {'name': 'mbps', 'op': np.sum},
+        'elapsed': {'name': 'elapsed-time', 'op': np.average},
+        'files': {'name': 'Files-per-thread', 'op': np.sum},
+        'files-per-sec': {'name': 'Files-per-sec', 'op': np.sum},
+        'records': {'name': 'Rec-per-thread', 'op': np.sum},
+    }
+
+    def __init__(self, uuid, crd):
+        """
+        Initialize the object by reading some of the data from the CRD file and
+        by connecting to the ES server and read all results from it.
+
+        Args:
+            uuid (str): the unique uid of the test
+            crd (dict): dictionary with test parameters - the test yaml file
+                        that modify it in the test itself.
+
+        """
+
+        self.uuid = uuid
+
+        self.server = crd['spec']['elasticsearch']['server']
+        self.port = crd['spec']['elasticsearch']['port']
+        self.index = crd['spec']['es_index'] + '-results'
+        self.new_index = crd['spec']['es_index'] + '-fullres'
+        self.all_results = {}
+
+        # make sure we have connection to the elastic search server
+        log.info(f'Connecting to ES {self.server} on port {self.port}')
+        try:
+            self.es = Elasticsearch([{'host': self.server, 'port': self.port}])
+        except ESExp.ConnectionError:
+            log.error('can not connect to ES server {}:{}'.format(
+                self.server, self.port))
+            raise
```

Comment:
@Avilir since we are going to run this on AWS as well, I think we should ignore here.

Reply:
@vasukulkarni This is not possible since the benchmark sending the results to ES - from within the pod, so without that we do not have full results.
we must have ES with public IP so the benchmark can send results to it, or we need to change the benchmark itself to collect and report the results in different way.

Reply:
we will need a mark to skip this test in AWS which can be taken as follow on PR